File: oc/scanview.m
//
//  scanview.m
//  scan
//  Copyright © 2016年 lixin. All rights reserved.
//

#import "scanview.h"
@interface scanview()
@property(nonatomic,assign)NSInteger picname;
@end
@implementation scanview

-(instancetype)initWithFrame:(CGRect)frame{
  self =  [super initWithFrame:frame];
    if (self) {
        self.backgroundColor = [UIColor clearColor];
        self.picname = 1;
    }
    return self;
}
-(void)drawRect:(CGRect)rect{
    NSString *str = [NSString stringWithFormat:@"%zi",_picname];
    UIImage *view = [UIImage imageNamed:str];
    [view drawInRect:self.bounds];
}
-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    UITouch *touch = [touches anyObject];
    CGPoint touchpoint = [touch locationInView:self];
    if (touchpoint.x < self.bounds.size.width / 2.0) {
        _picname -= 1;
        if (_picname < 1) {
            _picname = 6;
        }
    }
    if (touchpoint.x > self.bounds.size.width / 2.0) {
        _picname += 1;
        if (_picname > 6) {
            
            _picname = 1;
        }
    }
    [self setNeedsDisplay];
}
@end

File: swift/scanview.swift
//
//  scanview.swift
//  scan
//  Copyright © 2016年 lixin. All rights reserved.
//

import UIKit

class scanview: UIView {
    var picturename:Int = 1{
        didSet{
            setNeedsDisplay()
        }
    } //over
    override func drawRect(rect: CGRect) {
        let pic = UIImage(named:"\(picturename)")        
        guard (pic != nil) else{
            return
        }
        pic!.drawInRect(self.bounds)
    }//over
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        let touch = touches.first
        guard (touch != nil) else{
            return
        }
        let point = touch!.locationInView(self)
        if point.x < self.bounds.width / 2.0 {
            picturename -= 1
            if picturename < 1 {
                picturename = 1
            }
        }
        if point.x > self.bounds.width / 2.0 {
            picturename += 1
            if picturename > 6{
                picturename = 6
            }
        }
    }// over
}
